test/e2e/failover_test.go

```diff
@@ -242,6 +242,234 @@ var _ = framework.SerialDescribe("failover testing", func() {
 			})
 		})
 	})
+
+	ginkgo.Context("Application failover testing with purgeMode graciously", func() {
+		var policyNamespace, policyName string
+		var deploymentNamespace, deploymentName string
+		var deployment *appsv1.Deployment
+		var policy *policyv1alpha1.PropagationPolicy
+		var overridePolicy *policyv1alpha1.OverridePolicy
+		var maxGroups, minGroups int
+		ginkgo.BeforeEach(func() {
+			policyNamespace = testNamespace
+			policyName = deploymentNamePrefix + rand.String(RandomStrLength)
+			deploymentNamespace = testNamespace
+			deploymentName = policyName
+			deployment = testhelper.NewDeployment(deploymentNamespace, deploymentName)
+			maxGroups = 1
+			minGroups = 1
+
+			policy = &policyv1alpha1.PropagationPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: policyNamespace,
+					Name:      policyName,
+				},
+				Spec: policyv1alpha1.PropagationSpec{
+					ResourceSelectors: []policyv1alpha1.ResourceSelector{
+						{
+							APIVersion: deployment.APIVersion,
+							Kind:       deployment.Kind,
+							Name:       deployment.Name,
+						},
+					},
+					Placement: policyv1alpha1.Placement{
+						ClusterAffinity: &policyv1alpha1.ClusterAffinity{
+							ClusterNames: framework.ClusterNames(),
+						},
+						SpreadConstraints: []policyv1alpha1.SpreadConstraint{
+							{
+								SpreadByField: policyv1alpha1.SpreadByFieldCluster,
+								MaxGroups:     maxGroups,
+								MinGroups:     minGroups,
+							},
+						},
+					},
+					PropagateDeps: true,
+					Failover: &policyv1alpha1.FailoverBehavior{
+						Application: &policyv1alpha1.ApplicationFailoverBehavior{
+							DecisionConditions: policyv1alpha1.DecisionConditions{
+								TolerationSeconds: pointer.Int32(60),
```

Comment:
Can this parameter be set to a short value? For E2E, the waiting time may be long.

Same as `GracePeriodSeconds` value.

Reply:
Is 10 seconds enough?

Reply:
I feel like a little short. We need to wait the deployment running first. I feel like 30s is better.